src/bci_build/package/nginx/docker-entrypoint.sh

```diff
@@ -44,4 +44,21 @@ if [ "$1" = "nginx" ] || [ "$1" = "nginx-debug" ]; then
     fi
 fi
 
+CURRENT_UID=$(id -u)
+if [ "$CURRENT_UID" -gt "0" ]; then
+    # Running as Unprivileged User
+    entrypoint_log "$0: Running as unprivileged user (UID: $CURRENT_UID). Configuring for unprivileged mode (Port 8080)."
+
+    # Remove the 'user' directive
+    sed -i '/^user/d' /etc/nginx/nginx.conf
+    entrypoint_log "$0: Removed 'user' directive for unprivileged worker."
+
+    # Ensure PID path is set to /tmp/nginx.pid
+    sed -i 's,^#\?\s*pid\s\+.*;$,pid /tmp/nginx.pid;,' /etc/nginx/nginx.conf
+    sed -i "/^http {/a \        proxy_temp_path /tmp/proxy_temp;\n        client_body_temp_path /tmp/client_temp;\n        fastcgi_temp_path /tmp/fastcgi_temp;\n        uwsgi_temp_path /tmp/uwsgi_temp;\n        scgi_temp_path /tmp/scgi_temp;\n" /etc/nginx/nginx.conf
```

Comment:
Would it be better if we made such paths default for all users? Since it is `/tmp` anyway.